Add timepicker helper with added expectation to reduce flakiness

mkllnk opened the pull request:
#### What? Why?

- Closes #12786 <!-- Insert issue number here. -->

<!-- Explain why this change is needed and the solution you propose.
     Provide context for others to understand it. -->



#### What should we test?
<!-- List which features should be tested and how.
     This can be similar to the Steps to Reproduce in the issue.
     Also think of other parts of the app which could be affected
     by your change. -->

- Specs only.
- 100/100 passes locally.
- 140/140 passed on CI.

#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.


#### Dependencies
<!-- Does this PR depend on another one?
     Add the link or remove this section. -->



#### Documentation updates
<!-- Are there any wiki pages that need updating after merging this PR?
     List them here or remove this section. -->

mkllnk reviewed (commented):
This may still fail if CI is having a bad day and is very slow but then we can just increase the sleep time. It's only a fraction of a second and not noticeable by a human at the moment.

dacook commented on spec/support/features/datepicker_helper.rb:
```diff
@@ -45,6 +45,14 @@ def select_datetime_from_datepicker(datetime)
+      #                     50ms ->  87% success
+      #                    100ms -> 100% success in 112 runs
+      # Let's double that to reduce flakiness even further.
+      sleep 0.2
```
It'd be great if we could find an expectation for when Flatpickr has updated. But in the meantime this is a very reasonable fix!

dacook commented:
Thank for fixing this, I hope this is the last we see of it!
I'm curious, how did you run the test 140 times on CI? Did you write a temporary custom script on your branch?

mkllnk commented:
> https://github.com/openfoodfoundation/openfoodnetwork/commit/c37bdebf8b2b1f573892364249e357ce1711152a#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R87

I changed the CI script to run only the one test case. The existing matrix for Knapsack ran that 14 times. Example:

* https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/12421759049/job/34682119972

I then created an empty commit locally and pushed to Github. That triggered 14 runs in my repo on the branch and 14 in the OFN repo on the pull request. That's 28 in total. I repeated that a few times and I easily got to 100 hundred.

In the end, I removed the commit above, I could have reverted it for transparency.
